pkg/constants/constants.go

```diff
@@ -372,8 +372,6 @@ var (
 		autoscaling.MaxScaleAnnotationKey,
 		StorageInitializerSourceUriInternalAnnotationKey,
 		"kubectl.kubernetes.io/last-applied-configuration",
-		// remove when https://issues.redhat.com/browse/RHOAIENG-15662 is merged on community and ported to ODH
-		// Plus, this annotation must be moved to the inferenceservice-config
 		"security.opendatahub.io/enable-auth",
```

Comment:
My only observation about it is that it is better to keep it hardcoded for a while since if we remove it, upgrades might fail due to this value not being set if the reconciliation of the `inferenceservice-config` CM is disabled. Similar to what happened with the headless service recently.

Reply:
Without your pr, it conflicted a lot because the commits was based on your commit. You worked with verdant about this commits so you would know more about this than me. If you think it needs to be hardcoded please sent another commit to this pr or you can send a new pr.